feeds-lookup.js

```diff
@@ -230,94 +229,83 @@ exports.init = function (sbot, config) {
     }
   }
 
-  return {
-    loadState(cb) {
-      if (!loadStateRequested) loadState()
-
-      if (cb) stateLoadedP.promise.then(cb)
-    },
+  function findById(feedId, cb) {
+    try {
+      assertFeedId(feedId)
+      if (!validate.detectFeedFormat(feedId)) throw Error('Invalid feedId')
+    } catch (err) {
+      return cb(err)
+    }
 
-    ensureLoaded(feedId, cb) {
-      if (!loadStateRequested) loadState()
+    sbot.db.query(
+      where(subfeed(feedId)),
+      toCallback((err, msgs) => {
+        if (err) return cb(err)
 
-      if (detailsLookup.has(feedId)) cb()
-      else ensureQueue.add(feedId, cb)
-    },
+        msgs = msgs.filter((msg) => validate.isValid(msg))
+        if (msgs.find((m) => m.value.content.type === 'metafeed/tombstone')) {
+          return cb(null, null)
+        }
+        msgs = msgs.filter((m) =>
+          m.value.content.type.startsWith('metafeed/add/')
+        )
+        if (msgs.length === 0) {
+          return cb(null, null)
+        }
 
-    findByIdSync(feedId) {
-      if (!stateLoaded) {
-        throw new Error('Please call loadState() before using findByIdSync()')
-      }
-      assertFeedId(feedId)
+        const details = msgToDetails(undefined, msgs[0])
+        cb(null, details)
+      })
+    )
+  }
 
-      return detailsLookup.get(feedId)
-    },
+  function branchStream(opts) {
+    if (!loadStateRequested) {
+      loadState()
+      const deferred = Defer.source()
+      stateLoadedP.promise.then(() => {
+        deferred.resolve(branchStream(opts))
+      })
+      return deferred
+    }
```

Comment:
This whole if() is an important change in `branchStream`, it wasn't there before. All the rest should be the same.

This if() guarantees that the state is loaded before the stream can be consumed.

Reply:
yep :)